From 11d0cce62afc157468e1d97ea80a2510091ea2c2 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Fri, 1 Jul 2022 11:57:00 +0530
Subject: [PATCH] mac80211: Remove unused RX_FLAGS from mac80211_rx_flags

Remove unused RX_FLAG_AMPDU_DELIM_CRC_KNOWN flag from
mac80211_rx_flags to provide space for new EHT flags.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 include/net/mac80211.h | 33 +++++++++++++++------------------
 net/mac80211/rx.c      |  7 +------
 2 files changed, 16 insertions(+), 24 deletions(-)

--- a/include/net/mac80211.h
+++ b/include/net/mac80211.h
@@ -1410,8 +1410,6 @@ ieee80211_tx_info_clear_status(struct ie
  * @RX_FLAG_AMPDU_IS_LAST: this subframe is the last subframe of the A-MPDU
  * @RX_FLAG_AMPDU_DELIM_CRC_ERROR: A delimiter CRC error has been detected
  *	on this subframe
- * @RX_FLAG_AMPDU_DELIM_CRC_KNOWN: The delimiter CRC field is known (the CRC
- *	is stored in the @ampdu_delimiter_crc field)
  * @RX_FLAG_MIC_STRIPPED: The mic was stripped of this packet. Decryption was
  *	done by the hardware
  * @RX_FLAG_ONLY_MONITOR: Report frame only to monitor interfaces without
@@ -1468,39 +1466,38 @@ ieee80211_tx_info_clear_status(struct ie
  *	hardware or driver)
  */
 enum mac80211_rx_flags {
-	RX_FLAG_MMIC_ERROR		= BIT(0),
-	RX_FLAG_DECRYPTED		= BIT(1),
-	RX_FLAG_ONLY_MONITOR		= BIT(2),
-	RX_FLAG_MMIC_STRIPPED		= BIT(3),
-	RX_FLAG_IV_STRIPPED		= BIT(4),
-	RX_FLAG_FAILED_FCS_CRC		= BIT(5),
-	RX_FLAG_FAILED_PLCP_CRC 	= BIT(6),
+	RX_FLAG_MMIC_ERROR              = BIT(0),
+	RX_FLAG_DECRYPTED               = BIT(1),
+	RX_FLAG_ONLY_MONITOR            = BIT(2),
+	RX_FLAG_MMIC_STRIPPED           = BIT(3),
+	RX_FLAG_IV_STRIPPED             = BIT(4),
+	RX_FLAG_FAILED_FCS_CRC          = BIT(5),
+	RX_FLAG_FAILED_PLCP_CRC         = BIT(6),
 	/* one free bit at 7 */
-	RX_FLAG_NO_SIGNAL_VAL		= BIT(8),
-	RX_FLAG_AMPDU_DETAILS		= BIT(9),
-	RX_FLAG_PN_VALIDATED		= BIT(10),
-	RX_FLAG_DUP_VALIDATED		= BIT(11),
-	RX_FLAG_AMPDU_LAST_KNOWN	= BIT(12),
-	RX_FLAG_AMPDU_IS_LAST		= BIT(13),
-	RX_FLAG_AMPDU_DELIM_CRC_ERROR	= BIT(14),
-	RX_FLAG_AMPDU_DELIM_CRC_KNOWN	= BIT(15),
-	RX_FLAG_MACTIME			= BIT(16) | BIT(17),
-	RX_FLAG_MACTIME_PLCP_START	= 1 << 16,
-	RX_FLAG_MACTIME_START		= 2 << 16,
-	RX_FLAG_MACTIME_END		= 3 << 16,
-	RX_FLAG_SKIP_MONITOR		= BIT(18),
-	RX_FLAG_AMSDU_MORE		= BIT(19),
-	RX_FLAG_RADIOTAP_TLV_AT_END	= BIT(20),
-	RX_FLAG_MIC_STRIPPED		= BIT(21),
-	RX_FLAG_ALLOW_SAME_PN		= BIT(22),
-	RX_FLAG_ICV_STRIPPED		= BIT(23),
-	RX_FLAG_AMPDU_EOF_BIT		= BIT(24),
-	RX_FLAG_AMPDU_EOF_BIT_KNOWN	= BIT(25),
-	RX_FLAG_RADIOTAP_HE		= BIT(26),
-	RX_FLAG_RADIOTAP_HE_MU		= BIT(27),
-	RX_FLAG_RADIOTAP_LSIG		= BIT(28),
-	RX_FLAG_NO_PSDU			= BIT(29),
-	RX_FLAG_8023			= BIT(30),
+	RX_FLAG_NO_SIGNAL_VAL           = BIT(8),
+	RX_FLAG_AMPDU_DETAILS           = BIT(9),
+	RX_FLAG_PN_VALIDATED            = BIT(10),
+	RX_FLAG_DUP_VALIDATED           = BIT(11),
+	RX_FLAG_AMPDU_LAST_KNOWN        = BIT(12),
+	RX_FLAG_AMPDU_IS_LAST           = BIT(13),
+	RX_FLAG_AMPDU_DELIM_CRC_ERROR   = BIT(14),
+	RX_FLAG_MACTIME                 = BIT(15) | BIT(16),
+	RX_FLAG_MACTIME_PLCP_START      = 1 << 15,
+	RX_FLAG_MACTIME_START           = 2 << 15,
+	RX_FLAG_MACTIME_END             = 3 << 15,
+	RX_FLAG_SKIP_MONITOR            = BIT(17),
+	RX_FLAG_AMSDU_MORE              = BIT(18),
+	RX_FLAG_RADIOTAP_TLV_AT_END     = BIT(19),
+	RX_FLAG_MIC_STRIPPED            = BIT(20),
+	RX_FLAG_ALLOW_SAME_PN           = BIT(21),
+	RX_FLAG_ICV_STRIPPED            = BIT(22),
+	RX_FLAG_AMPDU_EOF_BIT           = BIT(23),
+	RX_FLAG_AMPDU_EOF_BIT_KNOWN     = BIT(24),
+	RX_FLAG_RADIOTAP_HE             = BIT(25),
+	RX_FLAG_RADIOTAP_HE_MU          = BIT(26),
+	RX_FLAG_RADIOTAP_LSIG           = BIT(27),
+	RX_FLAG_NO_PSDU                 = BIT(28),
+	RX_FLAG_8023                    = BIT(29),
 };
 
 /**
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -507,18 +507,13 @@ ieee80211_add_rx_radiotap_header(struct
 			flags |= IEEE80211_RADIOTAP_AMPDU_IS_LAST;
 		if (status->flag & RX_FLAG_AMPDU_DELIM_CRC_ERROR)
 			flags |= IEEE80211_RADIOTAP_AMPDU_DELIM_CRC_ERR;
-		if (status->flag & RX_FLAG_AMPDU_DELIM_CRC_KNOWN)
-			flags |= IEEE80211_RADIOTAP_AMPDU_DELIM_CRC_KNOWN;
 		if (status->flag & RX_FLAG_AMPDU_EOF_BIT_KNOWN)
 			flags |= IEEE80211_RADIOTAP_AMPDU_EOF_KNOWN;
 		if (status->flag & RX_FLAG_AMPDU_EOF_BIT)
 			flags |= IEEE80211_RADIOTAP_AMPDU_EOF;
 		put_unaligned_le16(flags, pos);
 		pos += 2;
-		if (status->flag & RX_FLAG_AMPDU_DELIM_CRC_KNOWN)
-			*pos++ = status->ampdu_delimiter_crc;
-		else
-			*pos++ = 0;
+		*pos++ = 0;
 		*pos++ = 0;
 	}
 
